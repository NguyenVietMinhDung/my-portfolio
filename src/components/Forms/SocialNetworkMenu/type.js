export type SocialNetworkItem = {
  name: string,
  url: string,
  iconUrl: string,
};

export type SocialNetworkMenuProps = {
  items: Array<SocialNetworkItem>,
};
