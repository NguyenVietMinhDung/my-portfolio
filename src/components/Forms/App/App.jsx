import React from 'react';
import Navbar from '../Navbar';

export default () => (
  <div className="container">
    <Navbar />
  </div>
);
