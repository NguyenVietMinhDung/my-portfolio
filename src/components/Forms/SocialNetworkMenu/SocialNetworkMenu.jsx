// @flow
import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './SocialNetworkMenu.scss';
import type { SocialNetworkMenuProps } from './type';
import SocialNetwork from './components/SocialNetworkItem';

const SocialNetworkMenu = (props: SocialNetworkMenuProps) => {
  const { items } = props;
  return (
    <ul styleName="social-network-menu">
      {items.map(item => (
        <SocialNetwork
          key={item.name}
          url={item.url}
          iconUrl={item.iconUrl}
        />
      ))}
    </ul>
  );
};

export default CSSModules(SocialNetworkMenu, styles);
