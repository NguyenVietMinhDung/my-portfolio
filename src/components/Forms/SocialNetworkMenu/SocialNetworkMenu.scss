.social-network-menu {
  display: flex;
  flex-flow: row wrap;
}