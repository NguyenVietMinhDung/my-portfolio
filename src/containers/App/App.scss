.loader {
  position: absolute;
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
}