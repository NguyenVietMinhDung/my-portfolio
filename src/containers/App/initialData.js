import facebookIconUrl from '../../assets/images/facebook.png';
import instagramIconUrl from '../../assets/images/instagram.png';
import linkedInIconUrl from '../../assets/images/linkedin.png';

export const navbarItems = [
  {
    name: 'Home',
    url: '#',
  },
  {
    name: 'About',
    url: '#',
  },
  {
    name: 'Services',
    url: '#',
  },
  {
    name: 'Portfolio',
    url: '#',
  },
  {
    name: 'Blog',
    url: '#',
  },
  {
    name: 'Contact',
    url: '#',
  },
];

export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];

export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/d%C5%A9ng-nguy%E1%BB%85n-502865114',
    iconUrl: linkedInIconUrl,
  },
];

export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 100,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS, Redux',
    value: 80,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS, Styled-Component',
    value: 80,
  },
  {
    itemId: 'skill-4',
    name: 'Jest, Enzyme',
    value: 70,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 80,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 50,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 80,
  },
];
