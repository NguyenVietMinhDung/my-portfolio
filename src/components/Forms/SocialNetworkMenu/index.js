import SocialNetworkMenu from './SocialNetworkMenu';

export default SocialNetworkMenu;
