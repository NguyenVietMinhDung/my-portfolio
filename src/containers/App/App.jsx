// @flow
import React, { memo } from 'react';
import type { AppProps } from './type';
import { navbarItems, socialNetworks } from './initialData';
import logoUrl from '../../assets/images/logo.png';
import hamburgerIconUrl from '../../assets/images/hamburger-icon.png';
import closeIconUrl from '../../assets/images/close-icon.png';
import avatarUrl from '../../assets/images/avatar.png';
import constants from '../../constants';
import Forms from '../../components/Forms';

const {
  Navbar,
  Header,
  AboutMe,
} = Forms;

const {
  header: {
    HEADER_NAME,
    HEADER_DESCRIPTION,
    HEADER_BUTTON_TEXT,
  },
  aboutMe: {
    ABOUT_ME_TITLE,
    ABOUT_ME_GREETING,
    ABOUT_ME_SUMMARY,
  },
} = constants;

const App = (props: AppProps) => (
  <>
    <Navbar
      navbarItems={navbarItems}
      logoUrl={logoUrl}
      hamburgerIconUrl={hamburgerIconUrl}
      closeIconUrl={closeIconUrl}
      {...props}
    />
    <Header
      name={HEADER_NAME}
      description={HEADER_DESCRIPTION}
      btnTxt={HEADER_BUTTON_TEXT}
      socialNetworks={socialNetworks}
      avatarUrl={avatarUrl}
    />
    <AboutMe
      title={ABOUT_ME_TITLE}
      greeting={ABOUT_ME_GREETING}
      summary={ABOUT_ME_SUMMARY}
    />
  </>
);

export default memo(App);
