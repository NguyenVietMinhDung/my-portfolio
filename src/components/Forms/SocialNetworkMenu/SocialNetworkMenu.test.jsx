import React from 'react';
import { shallow } from 'enzyme';
import { socialNetworks } from '../../../containers/App/initialData';
import SocialNetworkMenu from './SocialNetworkMenu';

const defaultProps = {
  items: socialNetworks,
};

describe('<SocialNetworkMenu />', () => {
  test('renders correctly', () => {
    const component = shallow(<SocialNetworkMenu {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
