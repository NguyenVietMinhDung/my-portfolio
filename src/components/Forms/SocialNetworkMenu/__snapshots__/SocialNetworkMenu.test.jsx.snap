// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialNetworkMenu /> renders correctly 1`] = `
<ul
  className="social-network-menu"
>
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="Facebook"
    url="https://www.facebook.com/minh.dung.391"
  />
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="Instagram"
    url="https://www.instagram.com/dungvmnguyen"
  />
  <Memo(SocialNetworkItem)
    iconUrl="test-file-stub"
    key="LinkedIn"
    url="https://www.linkedin.com/in/d%C5%A9ng-nguy%E1%BB%85n-502865114"
  />
</ul>
`;
